client/lib/domains/google-apps-users/index.js

```diff
@@ -37,7 +37,7 @@ function validate( { users, fields, domainSuffix } ) {
 				if ( field.value.length > 60 ) {
 					error = i18n.translate( 'This field can\'t be longer than 60 characters.' );
 				}
-			} else if ( 'email' === key ) {
+			} else if ( includes( [ 'email', 'username' ], key ) ) {
 				if ( /[^[0-9a-z_'.-]/i.test( field.value ) ) {
```

Comment:
Quick check: Do we lowercase these inputs somewhere? We should either allow capital letters or lowercase the input if we're not doing so already.